pkg/flowkit/gateway/emulator.go

```diff
@@ -19,31 +19,47 @@
 package gateway
 
 import (
-	"fmt"
-	"time"
-
-	"github.com/onflow/flow-cli/pkg/flowkit"
-	"github.com/onflow/flow-cli/pkg/flowkit/config"
-
-	emulator "github.com/onflow/flow-emulator"
-	flowGo "github.com/onflow/flow-go/model/flow"
+	"context"
 
 	"github.com/onflow/cadence"
+	emulator "github.com/onflow/flow-emulator"
+	"github.com/onflow/flow-emulator/server/backend"
 	"github.com/onflow/flow-go-sdk"
 	"github.com/onflow/flow-go-sdk/client"
 	"github.com/onflow/flow-go-sdk/client/convert"
+	flowGo "github.com/onflow/flow-go/model/flow"
+	"github.com/sirupsen/logrus"
+
+	"github.com/onflow/flow-cli/pkg/flowkit"
+	"github.com/onflow/flow-cli/pkg/flowkit/config"
 )
 
 type EmulatorGateway struct {
 	emulator *emulator.Blockchain
+	backend  *backend.Backend
+	ctx      context.Context
 }
 
 func NewEmulatorGateway(serviceAccount *flowkit.Account) *EmulatorGateway {
+	return NewEmulatorGatewayWithLogger(logrus.New(), serviceAccount)
+}
+
+func NewEmulatorGatewayWithLogger(logger *logrus.Logger, serviceAccount *flowkit.Account) *EmulatorGateway {
+	emulator := newEmulator(serviceAccount)
+	backend := backend.New(logger, emulator)
+	backend.EnableAutoMine()
+
 	return &EmulatorGateway{
-		emulator: newEmulator(serviceAccount),
+		emulator: emulator,
+		backend:  backend,
+		ctx:      context.Background(),
 	}
 }
 
+func (g *EmulatorGateway) SetContext(ctx context.Context) {
```

Comment:
Maybe not needed if with options is used

Reply:
I put this actually in line with my other comment, I mean I presumed most of the calls will not individually need context, but for example if you want to have timeout on let's say on script execution you can set a context before the call.